src/test/java/io/odpf/depot/message/proto/ProtoOdpfParsedMessageTest.java

```diff
@@ -347,7 +346,29 @@ public void shouldThrowExceptionIfColumnIsNotPresentInProto() throws IOException
         TestNestedMessageBQ nestedMessage = TestNestedMessageBQ.newBuilder().setNestedId("test").setSingleMessage(message1).build();
         ProtoOdpfParsedMessage protoOdpfParsedMessage = new ProtoOdpfParsedMessage(protoParser.parse(nestedMessage.toByteArray()));
         Assert.assertEquals("test", protoOdpfParsedMessage.getFieldByName("nested_id", odpfMessageSchema));
-        ConfigurationException configurationException = assertThrows(ConfigurationException.class, () -> protoOdpfParsedMessage.getFieldByName("single_message.order_id", odpfMessageSchema));
-        Assert.assertEquals("Invalid field config : single_message.order_id", configurationException.getMessage());
+        IllegalArgumentException exception = assertThrows(IllegalArgumentException.class, () -> protoOdpfParsedMessage.getFieldByName("single_message.order_id", odpfMessageSchema));
+        Assert.assertEquals("Invalid field config : single_message.order_id", exception.getMessage());
     }
+
+    @Test
+    public void shouldThrowExceptionIfColumnIsNotNested() throws IOException {
+        TestMessageBQ message1 = TestProtoUtil.generateTestMessage(now);
+        Parser protoParser = StencilClientFactory.getClient().getParser(TestNestedMessageBQ.class.getName());
+        OdpfMessageSchema odpfMessageSchema = odpfMessageParser.getSchema("io.odpf.depot.TestNestedMessageBQ", descriptorsMap);
+        TestNestedMessageBQ nestedMessage = TestNestedMessageBQ.newBuilder().setNestedId("test").setSingleMessage(message1).build();
+        ProtoOdpfParsedMessage protoOdpfParsedMessage = new ProtoOdpfParsedMessage(protoParser.parse(nestedMessage.toByteArray()));
+        Assert.assertEquals("test", protoOdpfParsedMessage.getFieldByName("nested_id", odpfMessageSchema));
+        IllegalArgumentException exception = assertThrows(IllegalArgumentException.class, () -> protoOdpfParsedMessage.getFieldByName("nested_id.order_id", odpfMessageSchema));
+        Assert.assertEquals("Invalid field config : nested_id.order_id", exception.getMessage());
+    }
+
+
+    @Test
+    public void shouldThrowExceptionIfFieldIsEmpty() throws IOException {
```

Comment:
let's add test case for repeated field as well?

Reply:
this test `shouldThrowExceptionIfColumnIsNotNested()`